src/runtime/recipe/recipe.ts

```diff
@@ -701,10 +701,9 @@ export class Recipe implements Cloneable<Recipe> {
     return slot;
   }
 
-  get isLongRunning(): boolean {
-    return this.triggers.some(group =>
-        group.some(trigger => trigger[0] === 'launch' && trigger[1] === 'startup')
-        && group.some(trigger => trigger[0] === 'arcId' && !!trigger[1]));
+  getTrigger(group: [string, string][], name: string): string | null {
+    const trigger = group.find(t => t[0] === name);
```

Comment:
Maybe to be clearer, we could destruct this? 
```
group.find(([key, _]) => key === name); 
```

Reply:
done